src/idoswitchmenuitem.c

```diff
@@ -224,6 +224,14 @@ ido_source_menu_item_state_changed (IdoActionHelper *helper,
                            g_variant_get_boolean (state));
 }
 
+static void ido_switch_menu_item_activate(IdoSwitchMenuItem *pSelf, gpointer pUserData)
```

Comment:
With the variable names you divert from what else is used in IDO (e.g. pUserData <-> user_data in ido_source_menu_item_state_changed().

Furthermore, please help me with my lack of C knowledge: how do the arguments play to gether with the g_signal_connect parameters below.?

Reply:
> Furthermore, please help me with my lack of C knowledge: how do the arguments play to gether with the g_signal_connect parameters below.?

**ido_action_helper_activate** does not send the current activation state of the switch. We need to connect to the menuitem's "activate" signal and call **ido_action_helper_activate_with_parameter** dynamically passing in the state of the switch. This is what the DBus client wants to know.

Reply:
> With the variable names you divert from what else is used in IDO (e.g. pUserData <-> user_data in ido_source_menu_item_state_changed().

Changed via https://github.com/AyatanaIndicators/ayatana-ido/pull/15/commits/7158a919a74eab737da674fabde1b2d39d967086